map.go

```diff
@@ -216,47 +180,81 @@ type MapKV struct {
 	Value interface{}
 }
 
+type mapInfo struct {
+	Type       MapType
+	KeySize    uint32
+	ValueSize  uint32
+	MaxEntries uint32
+	Flags      uint32
+}
+
+type IncompatibleMapError struct {
+	Existing mapInfo
+	Incoming mapInfo
+	diff     string
+}
+
+func (err *IncompatibleMapError) Error() string {
+	return err.diff
+}
+
 // Compatible returns nil if an existing map may be used instead of creating
 // one from the spec.
 //
-// Returns an error wrapping [ErrMapIncompatible] otherwise. The error also wraps [IncompatibleMapErr]
-// which contains useful information about the incompatibility.
+// If the Map is not compatible, returns an [IncompatibleMapError] wrapping
+// [ErrMapIncompatible]. The [IncompatibleMapError] contains the relevant fields
+// of the existing and incoming maps and can be inspected for error handling
+// using errors.As.
 func (ms *MapSpec) Compatible(m *Map) error {
 	ms, err := ms.fixupMagicFields()
 	if err != nil {
 		return err
 	}
 
-	asInfo := func(m *Map) *MapInfo {
-		return &MapInfo{
-			Type:       m.typ,
-			KeySize:    m.keySize,
-			ValueSize:  m.valueSize,
-			MaxEntries: m.maxEntries,
-			Flags:      m.flags,
-			Name:       m.name,
-		}
+	diffs := []string{}
+	if m.typ != ms.Type {
+		diffs = append(diffs, fmt.Sprintf("Type: %s -> %s", m.typ, ms.Type))
```

Comment:
Nit: use `instead of` instead of `->` and invert the order? The arrow doesn't really tell me more.

Reply:
What about `%v changed to %v`? `instead of` also completely lacks directionality, unlike an arrow indicator (apparently IMO).